node/overseer/overseer-gen/proc-macro/src/impl_builder.rs

```diff
@@ -107,46 +61,379 @@ pub(crate) fn impl_builder(info: &OverseerInfo) -> proc_macro2::TokenStream {
 		})
 		.collect::<Vec<_>>();
 
-	let generics = quote! {
-		< S, #( #baggage_generic_ty, )* >
-	};
-	let where_clause = quote! {
+	// A helper structure to carry fields of a builder to produce setters
+	struct BuilderFieldHelper {
+		// Used to distinguish baggage and subsystems
+		is_subsystem: bool,
+		// Valid for baggage as all subsystems are generics
+		is_generic: bool,
+		// Field name
+		name: Ident,
+		// Field type (either generic or not)
+		field_ty: TokenStream,
+		// Type to use in generics (like __FieldName)
+		placeholder_type: TokenStream,
+		// Used to constrain setter generics
+		where_clause: Option<TokenStream>,
+		// Specific initialised type, like OverseerFieldInit<T>
+		inited_ty: TokenStream,
+		// Specific uninitialised type, like OverseerFieldUninit<T>
+		uninited_ty: TokenStream,
+		// Order of the field in a target structure
+		order: usize,
+	}
+
+	let mut field_helpers = info.subsystems()
+		.iter()
+		.filter(|ssf| !ssf.wip)
+		.enumerate()
+		.map(|(idx, ssf)| {
+			let prefixed_ident = format_ident!("__{}",
+				ssf.name.to_string().to_class_case());
```

Comment:
This is only necessary due to the abstraction chosen. If we can avoid this call, we can avoid an entire dependency package.

Reply:
This is added to improve error messages (so they include the violated field name) and avoiding snake_case types.